### PR TITLE
CLOUD-272 HDI - Script action failure on Hive cluster due to unravel_es dir missing and wrong check for process

### PR DESCRIPTION
**Pull request by alejandrounravel**
### Summary
HDI - Script action failure on Hive cluster due to unravel_es dir missing and wrong check for process

### Testing
Already confirmed it works on an existing cluster.
Screen Shot 2019-03-21 at 4 59 21 PM
```
Unravel MR Sensor (unravel_es) is installed and running
Performing Unravel ES post-install checks to ensure service is running and properly configured
ES Process = /bin/bash ./dbin/unravel_emr_sensor.sh ... /dbin/unravel_emr_sensor.sh

The unravel_es.properties file contains the following cluster-id=hv21sh
The unravel_es.properties file contains the following unravel-server=10.10.1.93
The Unravel IP is properly formed
Done calling es_postinstall_check with result 0 (1 == failed, 0 == passed).
```

Also verified on a brand new cluster deployed. See internal Jira for details.

### Jira
https://unraveldata.atlassian.net/browse/CLOUD-272

### CODE REVIEW

File: hdi/hdinsight-unravel-spark-script-action/unravel_hdi_spark_bootstrap_4.5.sh [hunk 1]
```diff
@@ -895,28 +900,40 @@ function es_setup() {
...
+  if isFunction can_install_es ; then
+     can_install_es
+     if [ 0 -ne $? ]; then
+       echo "Not running on a master node, so can skip the ES Post Install check" | tee -a ${OUT_FILE}
+       return 0
+     fi
+  fi
   # make sure that 'unravel_es' is running and using correct arguments
-  local es_cmd=$(ps aexo "command" | grep unravel-emr-sensor | grep -v grep)
+  local es_cmd=$(ps aexo "command" | grep -E "unravel.emr.sensor" | grep -v "grep")
```
Comment (alejandrounravel) [inline hdi/hdinsight-unravel-spark-script-action/unravel_hdi_spark_bootstrap_4.5.sh]: The name of the process is different now, actually has "_" instead.

File: hdi/hdinsight-unravel-spark-script-action/unravel_hdi_spark_bootstrap_4.5.sh [hunk 2]
```diff
@@ -1847,7 +1864,8 @@ function cluster_detect() {
   local full_host_name=$(hostname -f)
 
   echo "AMBARI_USR=$AMBARI_USR" | tee -a ${OUT_FILE}
-  echo "AMBARI_PWD=$AMBARI_PWD" | tee -a ${OUT_FILE}
+  # Should not log the Ambari password.
+  #echo "AMBARI_PWD=$AMBARI_PWD" | tee -a ${OUT_FILE}
```
Comment (alejandrounravel) [inline hdi/hdinsight-unravel-spark-script-action/unravel_hdi_spark_bootstrap_4.5.sh]: Don't print passwords.

File: hdi/hdinsight-unravel-spark-script-action/unravel_hdi_spark_bootstrap_4.5.sh [hunk 3]
```diff
@@ -749,6 +749,7 @@ EOF
 }
 
 function gen_sensor_properties() {
+  sudo mkdir -p /usr/local/unravel_es/etc
```
Comment (alejandrounravel) [inline hdi/hdinsight-unravel-spark-script-action/unravel_hdi_spark_bootstrap_4.5.sh]: This was the actual fix.

File: hdi/hdinsight-unravel-spark-script-action/unravel_hdi_spark_bootstrap_4.5.sh [hunk 4]
```diff
@@ -2069,15 +2087,70 @@ function can_install_es() {
...
-    if [ 0 -ne $? ]; then
-        echo "ERROR: 'unravel_es' for Qubole does not use cluster-id" | tee -a ${OUT_FILE}
-        ret=1
+################################################################################################
+# Test an IP address for validity:                                                             #
+# Usage:                                                                                       #
+#   valid_ip IP_ADDRESS                                                                        #
+# Will return 0 if valid, otherwise, 1                                                         #
+################################################################################################
+function valid_ip() {
```
Comment (Unravel-Andy) [inline hdi/hdinsight-unravel-spark-script-action/unravel_hdi_spark_bootstrap_4.5.sh]: customer may enable TLS on Unravel UI, also need to support FQDN check

**Pull request merged**